Sources/NIOCore/AsyncSequences/NIOAsyncSequenceProducer.swift

```diff
@@ -229,12 +264,6 @@ extension NIOAsyncSequenceProducer {
             init(throwingSource: ThrowingSource) {
                 self._throwingSource = throwingSource
             }
-
-            @inlinable
-            deinit {
-                // We need to call finish here to resume any suspended continuation.
-                self._throwingSource.finish()
-            }
```

Comment:
In the throwing producer we call through to `sourceDeiniitialized` here -- why are they different?

Reply:
Because we store the throwing producer in here. So if we get deinited we will implicitly deinit the throwing one which will execute `sourceDeinitialized`.

Reply:
Gotcha, thanks for explaining. As we no longer need `deinit` can this become a `struct` then?